axiom/nr_txn.c

```diff
@@ -3519,3 +3521,24 @@ nr_php_package_t* nr_txn_add_php_package(nrtxn_t* txn,
   return nr_txn_add_php_package_from_source(txn, package_name, package_version,
                                             NR_PHP_PACKAGE_SOURCE_LEGACY);
 }
+
+void nr_txn_suggest_package_supportability_metric(nrtxn_t* txn,
+                                                  const char* package_name,
+                                                  const char* package_version) {
+  nr_php_package_t* p = NULL;
+
+  if (nrunlikely(NULL == txn)) {
+    return;
+  }
+
+  if (nr_strempty(package_name)) {
+    return;
+  }
+
+  nrl_verbosedebug(NRL_TXN, "Suggesting package %s %s", NRSAFESTR(package_name),
+                   NRSAFESTR(package_version));
+
+  p = nr_php_package_create_with_source(package_name, package_version,
+                                        NR_PHP_PACKAGE_SOURCE_SUGGESTION);
+  nr_php_packages_add_package(txn->php_package_suggestions, p);
+}
```

Comment:
What should this function do when `package_version` is `NULL`?

Reply:
FYI: `agent/tests/test_lib_aws_sdk_php.c` is testing current behavior [here](https://github.com/newrelic/newrelic-php-agent/blob/4ea22b6bc5d62439e690a5e2885e318a0b9b1a84/agent/tests/test_lib_aws_sdk_php.c#L124-L147).

Reply:
> What should this function do when `package_version` is `NULL`? It currently creates a package with `PHP_PACKAGE_VERSION_UNKNOWN` as suggested version.

That seems appropriate.

Reply:
My intention was to have `NULL` mapped to `PHP_PACKAGE_VERSION_UNKNOWN`.  This makes its use easier because many places in the instrumentation we get `NULL` for the version and not having to transform it in each place seems a better mechanism.  This way the version can just be passed into this function and it is known it will be handled properly.